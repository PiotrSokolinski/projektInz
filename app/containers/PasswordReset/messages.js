/*
 * PasswordReset Messages
 *
 * This contains all the text for the PasswordReset container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.PasswordReset'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Set a new password',
  },
  passwordEmptyError: {
    id: `${scope}.passwordEmptyError`,
    defaultMessage: 'Password is required',
  },
  passwordsDoNotMatchError: {
    id: `${scope}.passwordsDoNotMatchError`,
    defaultMessage: 'Passwords must match',
  },
  passwordConfirmEmpty: {
    id: `${scope}.passwordConfirmEmpty`,
    defaultMessage: 'Password confirm is required',
  },
  backToLogin: {
    id: `${scope}.backToLogin`,
    defaultMessage: 'Back to Login',
  },
  passwordLabel: {
    id: `${scope}.passwordLabel`,
    defaultMessage: 'Password',
  },
  repeatedPasswordLabel: {
    id: `${scope}.repeatedPasswordLabel`,
    defaultMessage: 'Retype password',
  },
  resetButton: {
    id: `${scope}.resetButton`,
    defaultMessage: 'Reset password',
  },
})
