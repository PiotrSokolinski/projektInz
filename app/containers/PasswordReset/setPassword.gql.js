import gql from 'graphql-tag'

export default gql`
  mutation setPasswordMutation($resetPasswordToken: String!, $newPassword: String!) {
    setPassword(resetPasswordToken: $resetPasswordToken, newPassword: $newPassword) {
      success
    }
  }
`
