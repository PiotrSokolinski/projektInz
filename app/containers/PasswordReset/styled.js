import styled from 'styled-components'
import Button from 'components/Button'
import { Link as RouterLink } from 'react-router-dom'
import { Colors } from 'themes'

export const ResetButton = styled(Button)`
  width: 100%;
`

export const Container = styled.div`
  display: flex;
  flex-directiom: column;
  justify-content: center;
  align-items: center;
`

export const Box = styled.div`
  padding: 40px;
  background: rgba(0, 0, 0, 0.8);
  box-sizing: border-box;
  box-shadow: 10px 10px 10px -5px rgba(0, 0, 0, 0.75);
  border-radius: 10px;
`

export const Header = styled.div`
  color: ${Colors.white};
  font-size: 30px;
  margin: 0 0 30px;
  padding: 0;
`

export const Link = styled(RouterLink)`
  float: right;
  text-decoration: none;
  color: ${Colors.white};
  font-size: 12px;
  line-height: 20px;
  transition: 0.5s;
  align-self: center;
  margin-top: 15px;
  &:hover {
    color: ${Colors.cerulean};
    transform: scale(1.1);
  }
`

export const ButtonsContainer = styled.div`
  display: flex;
  flex-direction: column;
`
