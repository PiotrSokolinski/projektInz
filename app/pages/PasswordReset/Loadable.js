/**
 * Asynchronously loads the component for PasswordResetPage
 */

import loadable from 'utils/loadable'

export default loadable(() => import('./index'))
