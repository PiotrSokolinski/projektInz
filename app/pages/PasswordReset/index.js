import React from 'react'
import PasswordResetContainer from 'containers/PasswordReset'
const PasswordReset = props => <PasswordResetContainer {...props} />

export default PasswordReset
