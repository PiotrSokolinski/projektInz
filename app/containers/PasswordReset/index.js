/**
 *
 * PasswordReset
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import * as Yup from 'yup'
import { FormattedMessage, injectIntl } from 'react-intl'
import { Formik } from 'formik'
import PublicInput from 'components/PublicInput'
import messages from './messages'
import * as Styled from './styled'

const initialValues = { password: '', repeatedPassword: '' }

const validationSchema = ({ intl }) =>
  Yup.object().shape({
    password: Yup.string().required(intl.formatMessage(messages.passwordEmptyError)),
    repeatedPassword: Yup.string()
      .oneOf([Yup.ref('password'), null], intl.formatMessage(messages.passwordsDoNotMatchError))
      .required(intl.formatMessage(messages.passwordConfirmEmpty)),
  })

const PasswordReset = ({ intl }) => (
  <Styled.Container>
    <Styled.Box>
      <Styled.Header>
        <FormattedMessage {...messages.header} />
      </Styled.Header>
      <Formik onSubmit={() => {}} initialValues={initialValues} validationSchema={validationSchema({ intl })}>
        {({ errors, touched, values, handleChange, handleBlur, handleSubmit }) => (
          <React.Fragment>
            <PublicInput
              id="passwordField"
              label={intl.formatMessage(messages.passwordLabel)}
              error={touched.password && errors.password}
              inputProps={{
                type: 'password',
                name: 'password',
                value: values.password,
                onChange: handleChange,
                onBlur: handleBlur,
              }}
            />
            <PublicInput
              id="repeatedPasswordField"
              label={intl.formatMessage(messages.repeatedPasswordLabel)}
              error={touched.repeatedPassword && errors.repeatedPassword}
              inputProps={{
                type: 'password',
                name: 'repeatedPassword',
                value: values.repeatedPassword,
                onChange: handleChange,
                onBlur: handleBlur,
              }}
            />
            <Styled.ButtonsContainer>
              <Styled.ResetButton type="submit" onClick={handleSubmit}>
                <FormattedMessage {...messages.resetButton} />
              </Styled.ResetButton>
              <Styled.Link to="/login">
                <FormattedMessage {...messages.backToLogin} />
              </Styled.Link>
            </Styled.ButtonsContainer>
          </React.Fragment>
        )}
      </Formik>
    </Styled.Box>
  </Styled.Container>
)

PasswordReset.propTypes = {
  intl: PropTypes.object,
}

export default injectIntl(PasswordReset)
