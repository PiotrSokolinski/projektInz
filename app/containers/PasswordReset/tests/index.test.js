/**
 *
 * Tests for PasswordReset
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import { MemoryRouter } from 'react-router-dom'

import PasswordReset from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<PasswordReset />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    const dispatch = jest.fn()
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <PasswordReset dispatch={dispatch} />
        </MemoryRouter>
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <PasswordReset />
        </MemoryRouter>
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
