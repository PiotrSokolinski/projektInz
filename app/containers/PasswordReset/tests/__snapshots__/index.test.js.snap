// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordReset /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-yjfqfl-1 UmXoV"
>
  <div
    class="styled__Box-yjfqfl-2 bzIPdi"
  >
    <div
      class="styled__Header-yjfqfl-3 cbooOk"
    >
      <span>
        Set a new password
      </span>
    </div>
    <div
      class="styled__InputBox-sc-1nwnwj1-1 iARNcM"
    >
      <div
        class="styled__Wrapper-sc-1nwnwj1-4 gJixhu"
      >
        <input
          class="styled__Input-sc-1nwnwj1-2 dfHhpa"
          name="password"
          type="password"
          value=""
        />
        <label
          class="styled__Label-sc-1nwnwj1-3 cYwlur"
        >
          Password
        </label>
      </div>
    </div>
    <div
      class="styled__InputBox-sc-1nwnwj1-1 iARNcM"
    >
      <div
        class="styled__Wrapper-sc-1nwnwj1-4 gJixhu"
      >
        <input
          class="styled__Input-sc-1nwnwj1-2 dfHhpa"
          name="repeatedPassword"
          type="password"
          value=""
        />
        <label
          class="styled__Label-sc-1nwnwj1-3 cYwlur"
        >
          Retype password
        </label>
      </div>
    </div>
    <div
      class="styled__ButtonsContainer-yjfqfl-5 jemEBa"
    >
      <button
        class="styled__ResetButton-yjfqfl-0 bflPZk styled__StyledButton-sc-1gb27vo-0 MZlKF"
        type="submit"
      >
        <span>
          Reset password
        </span>
      </button>
      <a
        class="styled__Link-yjfqfl-4 lmmxOt"
        href="/login"
      >
        <span>
          Back to Login
        </span>
      </a>
    </div>
  </div>
</div>
`;
